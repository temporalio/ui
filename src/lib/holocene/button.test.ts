import { afterEach, beforeEach, describe, expect, test } from 'vitest';

import Button from '$lib/holocene/button.svelte';

let host: HTMLElement;

beforeEach(() => {
  host = document.createElement('div');
  host.setAttribute('id', 'host');
  document.body.appendChild(host);
});

afterEach(() => {
  host.remove();
});

describe('Button', () => {
  test('default', async () => {
    const instance = new Button({ target: host, props: {} });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });

  test('disabled', async () => {
    const instance = new Button({
      target: host,
      props: {
        disabled: true,
      },
    });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });

  test('secondary', async () => {
    const instance = new Button({
      target: host,
      props: {
        secondary: true,
      },
    });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });

  test('secondary disabled', async () => {
    const instance = new Button({
      target: host,
      props: {
        secondary: true,
      },
    });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });

  test('loading', async () => {
    const instance = new Button({
      target: host,
      props: {
        loading: true,
      },
    });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });

  test('icon', async () => {
    const instance = new Button({
      target: host,
      props: {
        icon: 'download',
      },
    });

    expect(instance).toBeTruthy();

    expect(host.innerHTML).toMatchSnapshot();
  });
});
