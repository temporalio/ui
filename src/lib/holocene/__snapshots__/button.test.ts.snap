// Vitest Snapshot v1

exports[`Button > default 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\">  </button>"`;

exports[`Button > disabled 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\" disabled=\\"\\">  </button>"`;

exports[`Button > icon 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\"><span><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><title></title><path d=\\"M11.625 16.375C11.7188 16.4688 11.8438 16.5 12 16.5C12.125 16.5 12.25 16.4688 12.3438 16.375L16.8438 11.875C17.0312 11.6875 17.0312 11.3438 16.8438 11.1562C16.6562 10.9688 16.3125 10.9688 16.125 11.1562L12.5 14.8125V5.5C12.5 5.25 12.25 5 12 5C11.7188 5 11.5 5.25 11.5 5.5V14.8125L7.84375 11.1562C7.65625 10.9688 7.3125 10.9688 7.125 11.1562C6.9375 11.3438 6.9375 11.6875 7.125 11.875L11.625 16.375ZM17.5 18H6.5C6.21875 18 6 18.25 6 18.5C6 18.7812 6.21875 19 6.5 19H17.5C17.75 19 18 18.7812 18 18.5C18 18.25 17.75 18 17.5 18Z\\" fill=\\"currentcolor\\"></path></svg></span>  </button>"`;

exports[`Button > loading 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\"><span class=\\"animate-spin\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><title></title><path d=\\"M7.375 6.40625C6.84375 6.40625 6.375 6.875 6.375 7.40625C6.375 7.96875 6.84375 8.40625 7.375 8.40625C7.9375 8.40625 8.375 7.96875 8.375 7.40625C8.375 6.875 7.9375 6.40625 7.375 6.40625ZM7.375 15.625C6.84375 15.625 6.375 16.0625 6.375 16.625C6.375 17.1562 6.84375 17.625 7.375 17.625C7.9375 17.625 8.375 17.1562 8.375 16.625C8.375 16.0625 7.9375 15.625 7.375 15.625ZM12 17.5C11.4375 17.5 11 17.9688 11 18.5C11 19.0625 11.4375 19.5 12 19.5C12.5312 19.5 13 19.0625 13 18.5C13 17.9688 12.5312 17.5 12 17.5ZM5.5 11C4.9375 11 4.5 11.4688 4.5 12C4.5 12.5625 4.9375 13 5.5 13C6.03125 13 6.5 12.5625 6.5 12C6.5 11.4688 6.03125 11 5.5 11ZM12 4.5C11.4375 4.5 11 4.96875 11 5.5C11 6.0625 11.4375 6.5 12 6.5C12.5312 6.5 13 6.0625 13 5.5C13 4.96875 12.5312 4.5 12 4.5ZM18.5 11C17.9375 11 17.5 11.4688 17.5 12C17.5 12.5625 17.9375 13 18.5 13C19.0312 13 19.5 12.5625 19.5 12C19.5 11.4688 19.0312 11 18.5 11ZM16.5938 15.625C16.0312 15.625 15.5938 16.0625 15.5938 16.625C15.5938 17.1562 16.0312 17.625 16.5938 17.625C17.125 17.625 17.5938 17.1562 17.5938 16.625C17.5938 16.0625 17.125 15.625 16.5938 15.625Z\\" fill=\\"currentcolor\\"></path></svg></span>  </button>"`;

exports[`Button > secondary 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\">  </button>"`;

exports[`Button > secondary disabled 1`] = `"<button class=\\"button primary undefined svelte-bvxuak\\" type=\\"button\\">  </button>"`;
